### PR TITLE
feat(mini-nn): add neural network

### PR DESCRIPTION
**Pull request by rhtruong**
## What I changed

- Created NN model
- Train/validation/test data split

## Why I changed it

- Testing out LeakyReLU, Batch Normalization to every layer, mini-batch training influence on model accuracy

## How did I test it

-Ran pipeline with 150 epochs 

Results: 
Early stopping at epoch 18
Best Model found at Epoch: 3
Final Test Accuracy:       0.5457 (54.6%)
Final Test F1 Score:       0.5263
Final Test AUC-ROC:        0.5599

## Related Issues

---
**Note**: The PR title will become the commit message when squash merged, so make sure it follows conventional commit format!

### CODE REVIEW

**Review by JonathanPLev — APPROVED** [review]
nice! good work.
